src/averageAges.js

```diff
@@ -55,10 +63,15 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const withMother = people
+    .filter(person => onlyWithSon
+      ? people.some(child => child.name === person.mother) && person.sex === 'm'
```

Comment:
The same, DRY
Move it to the separate function
```suggestion
      ? ... && person.sex === 'm'
```